crates/relayer/src/chain/cosmos.rs

```diff
@@ -707,8 +711,10 @@ impl CosmosSdkChain {
         let key_pair = self.key()?;
         let key_account = key_pair.account();
 
-        let account =
-            get_or_fetch_account(&self.grpc_addr, &key_account, &mut self.account).await?;
+        // Take a write lock on the current account,
+        // excluding anybody from sending txs until we are done.
+        let mut opt_account = self.account.write().await;
```

Comment:
We take the lock here for the remaining duration of the function, ie. until the tx has been submitted.